packages/core/src/modules/vc/data-integrity/signature-suites/ed25519/Ed25519Signature2020.ts

```diff
@@ -0,0 +1,239 @@
+   * @param {Function} [options.verifier] - Verifier function that returns
+   *   an object with an async `verify()` method. Useful when working with a
+   *   KMS-provided verifier function.
+   *
+   * Advanced optional parameters and overrides.
+   *
+   * @param {object} [options.proof] - A JSON-LD document with options to use
+   *   for the `proof` node. Any other custom fields can be provided here
+   *   using a context different from security-v2).
+   * @param {string|Date} [options.date] - Signing date to use if not passed.
+   * @param {boolean} [options.useNativeCanonize] - Whether to use a native
+   *   canonize algorithm.
+   */
+  public constructor(options: Ed25519Signature2020Options) {
+    super({
+      type: 'Ed25519Signature2020',
+      algorithm: 'EdDSA',
+      LDKeyClass: options.LDKeyClass,
+      contextUrl: ED25519_SUITE_CONTEXT_URL_2020,
+      key: options.key,
+      proof: options.proof,
+      date: options.date,
+      useNativeCanonize: options.useNativeCanonize,
+    })
+    this.requiredKeyType = 'Ed25519VerificationKey2020'
+  }
+
+  public async assertVerificationMethod(document: JsonLdDoc) {
+    if (!_includesCompatibleContext({ document: document })) {
+      // For DID Documents, since keys do not have their own contexts,
+      // the suite context is usually provided by the documentLoader logic
+      throw new TypeError(
+        `The '@context' of the verification method (key) MUST contain the context url "${this.contextUrl}".`
+      )
+    }
+
+    if (!_isEd2020Key(document)) {
+      const verificationMethodType = jsonld.getValues(document, 'type')[0]
+      throw new Error(
+        `Unsupported verification method type '${verificationMethodType}'. Verification method type MUST be 'Ed25519VerificationKey2020'.`
+      )
+    } else if (_isEd2020Key(document) && !_includesEd2020Context(document)) {
+      throw new Error(
+        `For verification method type 'Ed25519VerificationKey2020' the '@context' MUST contain the context url "${ED25519_SUITE_CONTEXT_URL_2020}".`
+      )
+    }
+
+    // ensure verification method has not been revoked
+    if (document.revoked !== undefined) {
+      throw new Error('The verification method has been revoked.')
+    }
+  }
+
+  public async getVerificationMethod(options: { proof: Proof; documentLoader?: DocumentLoader }) {
+    let verificationMethod = await super.getVerificationMethod({
+      proof: options.proof,
+      documentLoader: options.documentLoader,
+    })
+
+    // convert Ed25519VerificationKey2020 to Ed25519VerificationKey2018
+    if (_isEd2020Key(verificationMethod) && _includesEd2020Context(verificationMethod)) {
+      // -- convert multibase to base58 --
+      let publicKeyBuffer = MultiBaseEncoder.decode(verificationMethod.publicKeyMultibase).data
+      if ((verificationMethod.publicKeyMultibase as string).startsWith('z6Mk')) {
+        publicKeyBuffer = publicKeyBuffer.slice(2)
+      }
```

Comment:
We have better tools to convert from multibase to base58:

(from fingerprint is not the best name and we should change that, but it does mutlibase decoding and handles the prefix for multiple key types with differeing prefix lengths)

```
Key.fromFingerprint(verificationMethod.publicKeyMultibase).publicKeyBase58
```

Reply:
Thanks! What are your thoughts on whether we have to be defensive here and check for whether the multicodec is present? My memory of the specs lead me to go the defensive approach but going back to the spec (https://www.w3.org/community/reports/credentials/CG-FINAL-di-eddsa-2020-20220724/#ed25519verificationkey2020), it looks like multicodec is explicitly required now. I might just be remembering wrong or thinking of a different spec that was more vague.

For contrast, in ACA-Py, we are currently taking the defensive approach.

Reply:
Hmm I know they changed it for old verification methods, which I thought was not possible (in my assumption Ed25519VerificationKey2018 would ALWAYS use publicKeyBase58). So I'm not wondering whether we should have a way more generic approach to extract the key material from a verification method?